packages/flutter/lib/src/rendering/sliver_fixed_extent_list.dart

```diff
@@ -182,9 +186,9 @@ abstract class RenderSliverFixedExtentBoxAdaptor extends RenderSliverMultiBoxAda
     final double leadingScrollOffset = indexToLayoutOffset(itemExtent, firstIndex);
     final double trailingScrollOffset = indexToLayoutOffset(itemExtent, lastIndex + 1);
 
-    assert(firstIndex == 0 || childScrollOffset(firstChild) <= scrollOffset);
+    // assert(firstIndex == 0 || childScrollOffset(firstChild) <= scrollOffset);
```

Comment:
@Hixie so I did a version of what we talked about. But I didn't find any logic that re-shifts all the content up so the missing gap is no longer present (Not too sure how to accomplish it. If we altered the sliver constraints to reduce it somehow, we'd have to tweak all the index for offset logic to skip the missing ones). I'm not sure if you wanted me to create that logic. 

I left it as is for now with the last layout-able item in the middle, but the code seems to indicate that mitigation should have existed.

Reply:
Hm, maybe we haven't added it for RenderSliverFixedExtentBoxAdaptor. You want to apply a [scrollOffsetCorrection](https://master-docs-flutter-io.firebaseapp.com/flutter/rendering/SliverGeometry/scrollOffsetCorrection.html). If you search for that term in some of the other sliver layout code you should find the relevant code path. I'm surprised we haven't implemented it for RenderSliverFixedExtentBoxAdaptor though. cc @abarth who may have context.

Reply:
Ok will add. Closing this for now.